mne/io/neuralynx/neuralynx.py

```diff
@@ -0,0 +1,165 @@
+
+    @verbose
+    def __init__(self, fname, preload=False, verbose=None):
+        _soft_import("neo", "Reading NeuralynxIO files", strict=True)
+        from neo.io import NeuralynxIO
+
+        fname = _check_fname(fname, "read", True, "fname", need_dir=True)
+
+        logger.info(f"Checking files in {fname}")
+
+        # get basic file info
+        nlx_reader = NeuralynxIO(dirname=fname)
+
+        info = create_info(
+            ch_types="seeg",
+            ch_names=nlx_reader.header["signal_channels"]["name"].tolist(),
+            sfreq=nlx_reader.get_signal_sampling_rate(),
+        )
+
+        # find total number of samples per .ncs file (`channel`) by summing
+        # the sample sizes of all segments
+        n_segments = nlx_reader.header["nb_segment"][0]
+        block_id = 0  # assumes there's only one block of recording
+        n_total_samples = sum(
+            nlx_reader.get_signal_size(block_id, segment)
+            for segment in range(n_segments)
+        )
+        super(RawNeuralynx, self).__init__(
+            info=info,
+            last_samps=[n_total_samples - 1],
+            filenames=[fname],
+            preload=preload,
+        )
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a chunk of raw data."""
+        from neo.io import NeuralynxIO
+
+        nlx_reader = NeuralynxIO(dirname=self._filenames[fi])
+        bl = nlx_reader.read(lazy=True)
+        # TODO: This is massively inefficient -- we should not need to read *all*
+        # samples just to get the ones from `idx` channels and `start:stop` time span.
+        # KA: neuralynx stores continuous recording in smaller segments (read in as
+        # neo.Segment objects) below code tries to be more efficient it delays reading
+        # into memory (nlx_reader.read(lazy=True)) and then reads in the idx channels
+        # and specified times by passing these as arguments in
+        # neo.Segment.AnalogSignal.load()
+
+        sr = nlx_reader.header["signal_channels"]["sampling_rate"][0]
+
+        # check that every segment has 1 associated neo.AnalogSignal() object (not sure
+        # what multiple analogsignals per neo.Segment would mean)
+        assert sum([len(segment.analogsignals) for segment in bl[0].segments]) == len(
+            bl[0].segments
+        )
+
+        # just a wrapper for readability
+        def _find_first_last_segment(
+            segments, start_spl: int, stop_spl: int, sr: float
+        ):
+            # gather the start and stop times (in sec) for all signals in segments
+            # (assumes 1 signal per segment)
+            # shape = (n_segments, 2) where 2nd dim is (start_time, stop_time)
+            onst_offt = np.array(
+                [
+                    (signal.t_start.item(), signal.t_stop.item())
+                    for segment in segments
+                    for signal in segment.analogsignals
+                ]
+            )
```

Comment:
Does this mean that there can be gaps and/or overlaps?

For gaps there is a way to handle it (insert zeros) but for overlaps I'm not even sure what the right behavior would be

Reply:
Oh, good point. I'd assume no gaps/overlap as it should just be chunked representation of a continuous recording by the acquisition system (for some reason). That is, each segments' off time should be the next segments' start time etc. Our test dataset only has two segments (over a total of ~3 sec recording). I can perhaps test this on my local datasets which have on the order of 10's of segments.

Reply:
If it's assumed that they are contiguous then I'd just construct an array of shape `(n_samples,)` of `int` type that gives you the mapping from sample numbers into segment numbers in `__init__` and store it in `_raw_extras[0] = dict(segment_map=segment-map)` or so. Then use this to figure out which segments to read in `_read_segments_file`. If this isn't clear I can push a quick commit to show what I mean

Reply:
Thanks! Makes total sense. Will implement.